test/module.test.js

```diff
@@ -1762,3 +1762,66 @@ describe('generate with prefix strategy', () => {
     expect(canonicalLink?.getAttribute('href')).toBe('nuxt-app.localhost/en')
   })
 })
+
+describe('Configuration with locale fallback decision map', () => {
+  let nuxt
+
+  beforeAll(async () => {
+    const override = {
+      i18n: {
+        lazy: true,
+        langDir: 'lang/',
+        locales: [
+          {
+            code: 'es',
+            iso: 'es',
+            name: 'Spanish',
+            file: 'es.js'
+          },
+          {
+            code: 'en-GB',
+            iso: 'en-GB',
+            name: 'Britain english',
+            file: 'en-GB.js'
+          },
+          {
+            code: 'fr-FR',
+            iso: 'fr-FR',
+            name: 'Français',
+            file: 'fr-FR.js'
+          },
+          {
+            code: 'de',
+            iso: 'de',
+            name: 'Deutsch',
+            file: 'de.js'
+          }
+        ],
+        vueI18n: {
+          fallbackLocale: {
+            'fr-FR': ['en-GB', 'es'],
+            default: ['de']
+          }
+        }
+      }
+    }
+
+    const localConfig = loadConfig(__dirname, 'locale-decision-map-fallback', override, { merge: true })
+    nuxt = (await setup(localConfig)).nuxt
+  })
+
+  afterAll(async () => {
+    await nuxt.close()
+  })
+
+  test('should be fallback translation by respecting the decision map', async () => {
+    const html = await get('/fr-FR/fallback')
+    const dom = getDom(html)
+
+    for (let level = 0; level <= 3; level++) {
+      const translationNode = dom.querySelector('[data-test="level' + level + '"]')
+      expect(translationNode).not.toBeNull()
+      expect(translationNode.textContent).toEqual('level ' + level)
```

Comment:
I would prefer to remove the assert above and just do this here:

```js
      expect(translationNode?.textContent).toEqual('level ' + level)
```

since otherwise type-checking complains about `translationNode` being potentially null.

Reply:
Job's done